src/api/firebase.js

```diff
@@ -87,10 +87,56 @@ export async function updateItem(listId, itemId, checked) {
 	const listItemRef = doc(db, listId, itemId);
 	const listItemSnap = await getDoc(listItemRef);
 
+	// Retrieving values stored in firebase
 	let totalPurchases = listItemSnap.data().totalPurchases;
```

Comment:
i don't know how expensive the `.data()` method is, but you might consider using a destructuring statement here, e.g. 
```javascript
let {totalPurchases, dateLastPurchased, dateCreated, dateNextPurchased} = listItemSnap.data();
```

Reply:
This is very clean, thank you for showing us another way to extract these values!